cores/esp8266/Schedule.cpp

```diff
@@ -152,28 +166,31 @@ void run_scheduled_recurrent_functions ()
         fence = true;
     }
 
-    recurrent_fn_t* prev = nullptr;
-    recurrent_fn_t* current = rFirst;
+    auto prev = &rFirst;
+    // prevent scheduling of new functions during this run
+    auto stop = rLast;
 
-    while (current)
+    bool done;
+    do
     {
-        if (current->callNow && !current->mFunc())
+        done = current == stop;
+        const bool wakeupToken = current->wakeupToken && current->wakeupToken->load();
```

Comment:
I don't understand how this logic (i.e.: surrounding code, not just this single line) relates to the explanation in the .h. Please add an explanation describing how the wakeup/wakeupToken/etc pieces interact.

Reply:
I hope the latest commit clears this up sufficiently, if you don't think so, please just let me know. Were you asking for an additional comment in the cpp, or is it the one the h file?